### PR TITLE
Feature/195/resend verification email

### PR DESCRIPTION
**Pull request by katelovescode**
# 💅🏼 What issue does this fix?
#195 - users can resend the verification email after signing up

Also adds e2e tests for Signup

## 🍂 Before You Submit

* [X] Did you write tests?
* [X] Did you run `bundle exec rspec` from the root?
* [X] Did you run `bundle exec rails rswag` from the root?
* [X] Did you run `bundle exec rubocop` from the root?
* [X] Did you run `yarn lint` in `/client`?
* [X] Did you run `yarn test` in `/client`?
* [ ] Are your primary keys UUIDs on any new tables?

## 🧵 Steps to set up locally

You'll need to restart your yarn server or rebuild tailwind manually because of a very minor config change

## 🧳 Steps to test
- visit `/signup`
- Sign up for an account
- you should be redirected to the Confirmation Sent page
- you should receive a confirmation email
- click the "Click here to resend the confirmation email" link
- you should see an "Email resent!" success box (per Slack discussions w/ @rebeccakarasiktw this is how we want to display these because the message object in the Figma mockup isn't positionable; it's always at the top center, instead of under the resend link)
- you should receive another confirmation email

## 🕯 Caveats, concerns
We might eventually want to put this ConfirmationSent view with the "resend" link in another view that someone can get redirected to if they're trying to log in before confirming. @csprayregen we'll need a ticket for that.

### CODE REVIEW

File: client/src/i18n/locales/es/index.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
   "add": "Añade",
   "addToContacts": "a tus contactos.",
   "resendConfirmationEmail": "Haga clic aquí para reenviar el correo.",
+  "confirmationEmailResent": "Correo electrónico reenviado",
```
Comment (katelovescode) [inline client/src/i18n/locales/es/index.json:59]: cc @csprayregen

File: config/locales/es.yml [hunk 2]
```diff
@@ -15,6 +15,7 @@ es:
     messages:
       taken: "ya ha sido tomado"
       confirmation: "Error de confirmación"
+      confirmation_resent: "No se pudo reenviar la confirmación"
```
Comment (katelovescode) [inline config/locales/es.yml:18]: cc @csprayregen

File: spec/cypress/integration/signup.spec.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
+import faker from 'faker'
+import { createSelector } from '../utils'
+
+const { name, internet, phone, company } = faker
+const firstName = name.firstName()
+const fullName = name.findName(firstName)
+const email = internet.email(firstName)
+// faker only allows for length and memorable options on its internet password object, so I can't guarantee there will be a number as well, so I'm hardcoding this
+const password = 'testpass1'
```
Comment (arku) [inline spec/cypress/integration/signup.spec.js:9]: I ran into this issue when running the Cypress tests a few times. How about using `random.alphaNumeric(15)`?

File: spec/cypress/integration/signup.spec.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    cy.route({
+      method: 'POST',
+      url: '/signup'
+    }).as('signup')
+    cy.visit('/signup')
+    cy.get(createSelector('organization')).type(orgName)
+    cy.get(createSelector('name')).type(fullName)
+    cy.get(createSelector('greetingName')).type(firstName)
+    cy.get(createSelector('multiBusiness')).click()
+    cy.get(createSelector('yesMultiBusiness')).click()
```
Comment (arku) [inline spec/cypress/integration/signup.spec.js:27]: Nice.

File: spec/cypress/integration/signup.spec.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    cy.get(createSelector('phoneType')).click()
+    cy.get(createSelector('homePhone')).click()
+    cy.get('[data-cy="languageEs"]').parent().click()
```
Comment (arku) [inline spec/cypress/integration/signup.spec.js:30]: A minor thing. Can we use `createSelector` here?

**Review by arku — DISMISSED** [review]
LGTM :100:

File: client/src/Signup/ConfirmationSent.js [hunk 6]
```diff
@@ -21,15 +22,30 @@ function ListItem({ children }) {
...
+  id: PropTypes.string
 }
 
 const ConfirmationSent = ({ userEmail }) => {
   const { t } = useTranslation()
+  const { makeRequest } = useApiResponse()
+  const [resent, setResent] = useState(null)
+
+  const resendConfirmation = async () => {
+    const response = await makeRequest({
```
Comment (arku) [inline client/src/Signup/ConfirmationSent.js:35]: We should reset `resent` to `null` here.

File: client/src/i18n/locales/es/index.json (hunk 1 above)
Comment (csprayregen) [inline client/src/i18n/locales/es/index.json:59]:
```suggestion
  "confirmationEmailResent": "Correo reenviado",
```

File: spec/cypress/integration/signup.spec.js (hunk 5 above)
Comment (katelovescode) [reply spec/cypress/integration/signup.spec.js:30]: Whoops, missed it 👍

**Review by arku — APPROVED** [review]
LGTM 👍

**Pull request merged**